unix.c

```diff
@@ -469,7 +469,11 @@ enet_socket_get_option (ENetSocket socket, ENetSocketOption option, int * value)
     {
         case ENET_SOCKOPT_ERROR:
             len = sizeof (int);
+#ifndef __3DS__ //getsockopt is unreliable on 3DS
             result = getsockopt (socket, SOL_SOCKET, SO_ERROR, value, & len);
```

Comment:
Were you seeing this codepath get reached during your testing? I don't see anything that calls it.

Reply:
I haven't seen it reached, I just added in this fix for completeness once I noticed it. Removing it wouldn't cause any issues if you'd prefer keeping the original behavior.

Reply:
What exactly do you mean when you say "getsockopt() is unreliable"? Are you talking about `SO_ERROR` specifically or just any `getsockopt()` call may fail randomly?

Reply:
Specifically SO_ERROR, at least that's the only call that I've seen fail so far. Here's the related libctru issue if you're interested: https://github.com/devkitPro/libctru/issues/412

Reply:
OK, that will probably cause issues with moonlight-common-c, since we use SO_ERROR for non-blocking connects there too.

Reply:
Oh it definitely caused issues with moonlight-common-c, it was how I first learned about this bug. Good to know this isn't a problem in enet though. Reverted!